lib/hash.h

```diff
@@ -232,6 +232,22 @@ extern void *hash_lookup(struct hash *hash, void *data);
  */
 extern void *hash_release(struct hash *hash, void *data);
 
+/*
+ * Remove an element exactly from a hash table.
+ *
+ * The data pointer is compared directly rather than by compare function.
+ *
+ * hash
+ *    hash table to operate on
+ *
+ * data
+ *    data element to remove
+ *
+ * Returns:
+ *    the removed element if found, or NULL if not found
+ */
+extern void *hash_release_exact(struct hash *hash, void *data);
```

Comment:
The problem isn't removing the wrong entry, it's having 2 (or more) "equal" entries to begin with. We can't go mess with the hash code like this to work around a bug elsewhere.

Reply:
agreed.  We need to move over to a better data structure